### ISSUE TITLE
match an expression

### ISSUE THREAD

**Issue by gregwebs** [puffnfresh/roy#114]
match only takes a variable

``` javascript
let v = f a b c
match v
  case ...
```

I would like that variable to be automatically generated by an expression

``` javascript
match f a b c
  case ...
```

At least this limitation should be documented for now

**Comment by puffnfresh** [comment]
You should be able to do:

```
match (f a b c)
  case ...
```

Is that a problem?

**Comment by gregwebs** [comment]
yes, that does work

**Comment by puffnfresh** [comment]
Fixed in 8f68d3f46fde81d9ab4d501c1d949362bd2322cc
